File: Example Apps/Example ObjC/ConnectionTabBarController.m
//
//  ConnectionTabBarController.m
//  SmartDeviceLink-Example-ObjC
//
//  Created by Joel Fischer on 9/23/21.
//  Copyright © 2021 smartdevicelink. All rights reserved.
//

#import "ConnectionTabBarController.h"

#import "Preferences.h"

@interface ConnectionTabBarController () <UITabBarControllerDelegate>

@end

@implementation ConnectionTabBarController

- (void)viewDidLoad {
    [super viewDidLoad];

    self.delegate = self;
}

#pragma mark - UITabBarControllerDelegate

- (void)tabBar:(UITabBar *)tabBar didSelectItem:(UITabBarItem *)item {
    Preferences.sharedPreferences.lastUsedSegment = [tabBar.items indexOfObject:item];
}

@end

File: Example Apps/Example Swift/ConnectionTabBarController.swift
//
//  ConnectionTabBarController.swift
//  SmartDeviceLink-Example-Swift
//
//  Created by Joel Fischer on 9/23/21.
//  Copyright © 2021 smartdevicelink. All rights reserved.
//

import UIKit

class ConnectionTabBarController: UITabBarController {
    override func viewDidLoad() {
        super.viewDidLoad()

        delegate = self
        selectedIndex = AppUserDefaults.shared.lastUsedSegment
    }
}

extension ConnectionTabBarController: UITabBarControllerDelegate {
    override func tabBar(_ tabBar: UITabBar, didSelect item: UITabBarItem) {
        if let index = tabBar.items?.firstIndex(where: {$0 == item}) {
            AppUserDefaults.shared.lastUsedSegment = index
        }
    }
}
